### DISCUSSION TITLE
Make non-curated applications follow system theme

### DISCUSSION THREAD

**Discussion by GustavoPeredo** [elementary/granite#495] [Ideas]
Having the option to support a system-wide dark theme in Granite is a step in the right direction, but that prevents non Granite applications to follow the system theme :(. Unfortunately, this means that my and others' applications (although supporting a dark mode), won't follow the system theme.

![el2](https://user-images.githubusercontent.com/30641461/116797988-b6f74300-aac1-11eb-969c-0c2d811f74a2.png)
_Example of my app working in dark mode_
![el3](https://user-images.githubusercontent.com/30641461/116797991-b9f23380-aac1-11eb-82d9-a64ebf6f84b9.png)
_Example of my app working in dark mode_

I am in favor of a dbus call to make apps follow the system theme, but while that isn't a freedesktop specification, I'd like users to have a good experience with my app on elementaryOS.

Here are my solutions/questions:

* Is it possible to somehow only use Granite's library to make an Adwaita application follow the system theme?
* Would it be possible for the elementaryOS team to develop a library that only does that? Or maybe integrate it in libadwaita?
* Would it be possible for an application to look for a flag/config file/dbus entry in elementary that signalizes that the system is in dark/light mode?

I would like to know your opinions!

Gustavo

**Comment by meisenzahl** [comment]
> We're working with GNOME on a FreeDesktop initiative for a system-wide dark style, but it has not been accepted/implemented by GNOME yet. elementary and GNOME designers agree it is critical that such an effort is opt-in for apps, so apps will use the light style unless they update to follow the preference.

> We will not manually force apps to use a different stylesheet than what the app requests, so there's nothing to do for us here.

_Originally posted by @cassidyjames in https://github.com/elementary/stylesheet/issues/1008#issuecomment-830835177_

**Reply by GustavoPeredo** [reply]
Thanks for your reply,
Yes, I agree with both atatements. Do you mean that there is no way for my non-granite application to follow the system's dark/light theme?

**Comment by danirabbit** [comment]
If you don't want to include Granite in your app, the simplest way is you can listen to the gsettings key at `org.freedesktop.prefers-color-scheme`.

The gsettings key is mirrored by elementary settings daemon from the pantheon accountsservice extension. So you could also listen to accountsservice over dbus, which is what Granite currently does: https://github.com/elementary/granite/blob/master/lib/Widgets/Settings.vala

Since there is still some debate among the various platforms about how exactly to expose this setting, it may change in the future. The primary advantage of using Granite here is that we will keep Granite up to date with these changes so that your app will continue to function as intended automatically.

**Reply by GustavoPeredo** [reply]
Thank you very much! I'll probably do that on my app. What is your opinion on a library that only listens to org.freedesktop.prefers-color-scheme? Do you think is a good idea? (I'm thinking of doing it)
